docs/how-to/virtual_memory.rst

```diff
@@ -0,0 +1,150 @@
+.. meta::
+  :description: This chapter describes introduces Virtual Memory (VM) and shows
+                how to use it in AMD HIP.
+  :keywords: AMD, ROCm, HIP, CUDA, virtual memory, virtual, memory, UM, APU
+
+.. _virtual_memory:
+
+*******************************************************************************
+HIP Virtual Memory Management
+*******************************************************************************
+
+Managing memory management is important when creating high-performance applications in the HIP ecosystem. Both memory allocation and copy can result in bottlenecks, which can significantly impact performance. 
+
+The global memory allocation in HIP happens with the C style allocation function. This
+works fine for simple cases, but there is a problem, if your memory needs change. If you
+need to increase the size of your memory, you'll have to allocate a second larger buffer
+and copy the data to it, before you can free the original buffer, which increases overall
+memory usage and causes unnecessary ``memcpy`` calls. An other solution is to have a
+larger buffer allocated, than you initially need, which isn't an efficient way to handle
+resources and doesn't solve the issue of reallocation, when the extra buffer runs out.
+
+Virtual memory management is the solution to these memory management problems. It helps
+to reduce memory usages and unnecessary ``memcpy`` calls.
+
+.. _memory_allocation_virtual_memory:
+Memory allocation
+===============================================================================
+
+When you need a chunk of memory on the device, previously you called the ``hipMalloc``
+function to allocate and it returns a pointer, that is immediately usable after the call.
+When using virtual memory this is separated into multiple steps using the
+``hipMemCreate``, ``hipMemAddressReserve``, ``hipMemMap`` and ``hipMemSetAccess``
+functions. In this guide you'll learn what these functions do and how you can use them
+for your needs.
+
+Allocate physical memory
+-------------------------------------------------------------------------------
+
+The first step is allocate the physical memory itself with the ``hipMemCreate`` function.
+It take the size of the buffer, an ``unsigned long long`` for flags and a
+``hipMemAllocationProp``, which contains the properties of the properties of the of the
+memory to be allocated, like where is this memory physically located or what kind of
+shareable handles should be available. If the allocation is successful the function
+returns with a ``hipSuccess`` and the ``hipMemGenericAllocationHandle_t`` will represent
+a valid physical memory allocation. It is important to note that the size has to be
+aligned with the granularity appropriate for the properties of the allocation. You can
+use the ``hipMemGetAllocationGranularity`` function to query this information.
+
+.. code-block::cpp
```

Comment:
I don't see any of the code-blocks in the draft document. I think the syntax is wrong and is missing a space before "cpp"

Reply:
Fixed.